src/hardware/mouse/mouse.cpp

```diff
@@ -37,14 +37,260 @@
 
 CHECK_NARROWING();
 
-bool seamless_driver  = false;
-bool seamless_setting = false;
-
 static Bitu int74_ret_callback = 0;
 
 static MouseQueue &mouse_queue  = MouseQueue::GetInstance();
 static ManyMouseGlue &manymouse = ManyMouseGlue::GetInstance();
 
+// ***************************************************************************
+// GFX-related decision making
+// ***************************************************************************
+
+static struct {
+	bool fullscreen   = false; // if full screen mode is active
+	uint16_t clip_x   = 0;     // clipping = size of black border (one side)
+	uint16_t clip_y   = 0;
+
+	bool captured     = false; // if GFX was requested to capture mouse
+	bool visible      = false; // if GFX was requested to make cursor visible
+	bool raw_input    = false; // if GFX was requested to provide raw movements
+
+	bool has_focus    = false; // if our window has focus
+	bool gui_takeover = false; // if a GUI requested to take over the mouse
+	bool user_capture = false; // if user requested mouse to be captured
+
+	bool seamless     = false; // if seamless mouse integration is in effect
+	bool drop_events  = true;  // if we should drop mouse events
+
+	bool capture_on_click  = false; // if any button click should capture the mouse
+	bool capture_on_middle = false; // if middle button press should capture the mouse
+	bool release_on_middle = false; // if middle button press should release the mouse
+	bool toggle_on_hotkey  = false; // if hotkey should toggle mouse capture
+
+	bool cursor_outside   = true; // if mouse cursor is outside of drawing area
```

Comment:
Suggest boolean prefix or postfixes.

`is_captured`
`is_visible`
`input_is_raw`

(`has_focus` is good!)

`gui_has_taken_over`

`capture_was_requested`

`is_seamless`
`should_drop_events`
`should_capture_on_click`
`should_capture_on_middle`
`should_release_on_middle`
`should_toggle_capture_on_hotkey`

`cursor_is_outside`

Reply:
OK, changed names.